### ISSUE TITLE
save-window does not respect the settings of the options

### ISSUE THREAD

**Issue by ralf3u** [lxqt/screengrab#233]
when saving, the directory is not the same like the directory in the options: according to the options it is the user-file, but in the save-window it is the home-file;
if in the options-window the File name is changed to nothing, so without name, then nevertheless there is the user name in the save-window in the File name

##### Expected Behavior
save-window should respect the settings of the options

##### Current Behavior
save-window does not respect the settings of the options

##### Possible Solution
I think 3 things need to be done in the save-window:
1) no File type in the File name (there is already a special field for that called "File type")
2) File Name should be exactly like in the options
3) same directory like in the options

##### Steps to Reproduce (for bugs)
1. open ScreenGrab
2. click in the symbol-bar on the options-symbol
3. in the options-window: General: delete "screen"
4. in the options-window: click on "Save"
5. click in the symbol-bar on the save-symbol: compare the options-window with the save-window (the directory and the File Name)

##### Context
save-window should always follow the settings of the options

##### System Information
* Distribution & Version: Lubuntu (German version)
* LXQt: 0.15.0
* Qt Version: 5.14.2
* ScreenGrab: 2.0.1
* Qt: 5.14.2

**Comment by stefonarch** [comment]
True. If  no default file name is given it will ignore directory and save to `~/tmp.<extension>
`
But it's a very rare condition.

Solution could be that default filenname isn't allow to be empty.

**Comment by tsujan** [comment]
Frankly, I didn't understand a word of this report. What's "symbol-bar"? What's "General: delete screen"?!! Why do you use the word "symbol" so many times?

**Comment by ralf3u** [comment]
@tsujan
These are the answers to your question:

What is a symbol-bar?
From the top of a window: first comes the title-bar, then normally the menu-bar and just down below the symbol-bar, where symbols or rather icons are placed.

Why do you use the word "symbol" so many times?
I wanted to be as precise as possible.
Instead of: click in the symbol-bar on the options-symbol
I could have used: open the options

What's "General: delete screen"?
open the options, then click on "General" (the word placed on top left; I don't use the English version, so maybe it is an other word), then in the File name there is the name "screen";
now delete the word "screen", so that the File Name is empty
(I did that because I like to give my own names to a screenshot)

**Comment by stefonarch** [comment]
"screen" is the default file name in the settings.
I suspect there's  a lot of google translate stuff.

**Comment by ralf3u** [comment]
@stefonarch
"Solution could be that default filenname isn't allow to be empty."

If this is the case, then at least it would be good to realize those 2 points in the save-window:
1) no file type in the File name (there is already a special field for that called "File type")
2) the file name should be marked, like for example in LibreOffice, so it can be erased faster

**Comment by ralf3u** [comment]
@stefonarch

you changed the title of the issue from
"save-dialog does not respect the settings of the options"
to
"If default file name is empty "save to folder setting" is not respected"

but I think your title does not mention that also the File name is not respected
(because there is the user name in the File name)

**Comment by ralf3u** [comment]
@stefonarch
"Solution could be that default filenname isn't allow to be empty."

if in the options I activate the checkbox to have the date and time in the name, then there are still 2 mistakes:
- wrong directory
- user name in the File name

**Comment by tsujan** [comment]
> @tsujan, These are the answers to your question:

@ralf3u, if you want your comments to be read, don' make up terms; learn to use the real terms instead, like *toolbar* and *icon* (if I've correctly guessed what yo mean). We don't have time to decipher your comments.

Here, we use English to communicate with each other, not to confuse each other.

**Comment by ralf3u** [comment]
@tsujan
I wrote again the issue because I did too many mistakes; I'm sorry for that

**Comment by tsujan** [comment]
@ralf3u 

Now it's much better. Thanks.

Try to make your future comments as short as possible, so that the reader could understand them quickly.

**Comment by stefonarch** [comment]
> * user name in the File name

? 

I see this behaviour:

1) leave empty both   `default save directory `and` default file: Name` > saving to ~ as `..png `or `..jpg`
2) Set /tmp in `default save directory `  and leave empty `Name > saving to` ~/tmp.png`

This is the issue.

3) set both and all is perfect.

**Comment by ralf3u** [comment]
and there is another issue: a hyphen before the date:
go in the options, click on "Advanced", and then activate "Insert current date and time", then click in the options on "Save"

now press ctrl+s: in the File name there is a hyphen before the date;

it should be without hyphen before the date, because in the options there was no hyphen before the date;
I mean, if I really wanted to have a hyphen before the date, then I can enter it in options in "Advanced" in the Template

Should I open new issue for that?

**Comment by stefonarch** [comment]
> Should I open a new issue for that?

Please not.

This happens when filename is left empty, and could be solved by make this mandatory.
`screen-12-10-20.png` is the way all works fine.

**Comment by ralf3u** [comment]
this is a pity, because the names of my screenshots start with the date;
if one day you decide to delete the hyphen, then please delete also the file type in the file name, because then it is easier to add some words at the end of the file name
